### PR TITLE
[M5] Implementation agnostic networking abstraction

### PR DESCRIPTION
**Pull request by hanabi1224**
**Summary of changes**

This PR tries to define implementation agnostic traits for networking layer, transition frontier layer and the message processor that facilitates the communication between them.

The traits are extensible for graphql backend and libp2p backend to power both wasm node(s) that run in browser and cli node(s) that run on x86-64 and arm-64 CPU(s). It also makes it easy to mock the networking layer to unittest the `TransitionFrontier`

Changes introduced in this pull request:
- [x] `TransitionFrontier` trait
- [x] `NonConsensusNetworkingOps` trait
- [x] `NetworkMessageProcessor`
- [x] `NonConsensusNetworkingOps` impl for https://graphql.minaexplorer.com/ backend
- [x] A naive `TransitionFrontier` impl
- [x] Simple test UI

**Reference issue to close (if applicable)**

Closes  https://github.com/ChainSafe/mina-rs/issues/233

**Other information and links**

### CODE REVIEW

**Review by sudo-shashank — APPROVED** [review]

File: apps/wasm/mvp/Cargo.toml [hunk 1]
```diff
@@ -12,14 +12,18 @@ publish = false
...
+mina-serialization-types = {path = "../../../protocol/serialization-types"}
 
 anyhow = "1"
 async-trait = "0.1"
 base64 = "0.13"
 hex = "0.4"
 js-sys = "0.3"
+lazy_static = "1"
 libp2p = {version = "0.44", features = ["wasm-bindgen", "wasm-ext-websocket"]}
+once_cell = "1"
```
Comment (creativcoder) [inline apps/wasm/mvp/Cargo.toml:26]: do we need both lazy_static and once_cell here?

File: network/src/processor/js/graphql_api_v1.rs [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+        Default::default()
+    }
+
+    /// TODO: Doc
+    pub async fn poll_latest_once(&self) -> anyhow::Result<()> {
+        if let Some(block_responder) = &self.block_responder {
+            let blocks = query_latest_blocks(10).await?;
+            for b in blocks {
+                if let Ok(block) = fetch_block(b.block_height, b.state_hash.as_str()).await {
+                    _ = block_responder.send(block).await;
```
Comment (creativcoder) [inline network/src/processor/js/graphql_api_v1.rs:75]: add `?` at the end?

File: network/src/processor/processor_impl.rs [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+        self.nonconsensus_ops.read().await
+    }
+
+    /// Schedules event loops of all types of communications between [TransitionFrontier] and
+    /// [NonConsensusNetworkingOps].
+    pub async fn run(&self) {
+        // let block_receiver = &mut self.block_receiver;
+        // let transition_frontier = &mut self.transition_frontier;
+        // let query_block_request_receiver = &mut self.query_block_request_receiver;
+        // let nonconsensus_ops = &mut self.nonconsensus_ops;
```
Comment (creativcoder) [inline network/src/processor/processor_impl.rs:52]: remove these?

File: network/src/processor/js/graphql_api_v1.rs (hunk 2 above)
Comment (hanabi1224) [reply network/src/processor/js/graphql_api_v1.rs:75]: error should not break the loop here, use `log::error!` here instead

File: network/src/processor/processor_impl.rs (hunk 3 above)
Comment (hanabi1224) [reply network/src/processor/processor_impl.rs:52]: fixed

**Review by creativcoder — APPROVED** [review]

**Pull request merged**